# Using a Decision Tree Classifier algorithm to assign labels on unknown data

In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score

In [2]:
# Load the data into a pandas DataFrame
df_combined = pd.read_csv('C:/Users/smvan/CFFS-S23/CFFS-22-23/data/final/2022_2023_CFFS_Outcomes/Data_Labelled_Gather22-23_with_name' +
                 '.csv', usecols=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 15])

In [3]:
df_combined.head()

Category            Displayed Name   ProdId                  Description  \
0  Caliente            Black bean soup  R-54947      CAL|Black bean soup|AAD   
1  Caliente             Guacamole bowl  R-55154       CAL|Bowl|Guacamole|AAD   
2  Caliente    Havana Add Chicken Bowl  R-62018  CAL|Bowl|Havana|Add Chicken   
3  Caliente    Havana Add Scallop Bowl  R-64909  CAL|Bowl|Havana|Add Scallop   
4  Caliente    Havana Plant-based Bowl  R-35896  CAL|Bowl|Havana|Plant based   

      SalesGroup  Weight (g)  GHG Emission (g)  N lost (g)  \
0  GV - CALIENTE       405.0        642.872067    3.699893   
1  GV - CALIENTE       345.0        577.312317    3.412346   
2  GV - CALIENTE       430.0       1120.107243   15.532360   
3  GV - CALIENTE       390.0        742.333777    6.220997   
4  GV - CALIENTE       415.0        769.178688    2.802186   

   Freshwater Withdrawals (L)  Stress-Weighted Water Use (L)  \
0                       53.60                        1723.18   
1                      115.97                        3039.52   
2                      125.86                         764.45   
3                       83.43                         677.50   
4                       83.83                         673.26   

   GHG Emission (g) / 100g  N lost (g) / 100g  \
0                   158.73               0.91   
1                   167.34               0.99   
2                   260.49               3.61   
3                   190.34               1.60   
4                   185.34               0.68   

   Freshwater Withdrawals (L) / 100g  Stress-Weighted Water Use (L) / 100g  \
0                              13.23                                425.48   
1                              33.61                                881.02   
2                              29.27                                177.78   
3                              21.39                                173.72   
4                              20.20                                162.23   

  Combined Label  
0          Green  
1          Green  
2         Yellow  
3          Green  
4          Green

In [4]:
# Split the dataset into input features (X) and target variable (y)
X = df_combined[['GHG Emission (g)', 'N lost (g)', 'Freshwater Withdrawals (L)', 'Stress-Weighted Water Use (L)']]
y = df_combined['Combined Label']

In [5]:
# Split the dataset into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

In [6]:
# Select the machine learning algorithm and fit the model to the training data
clf = DecisionTreeClassifier()
clf.fit(X_train, y_train)

DecisionTreeClassifier()

In [7]:
# Use the trained model to make predictions on the test dataset
y_pred = clf.predict(X_test)

In [8]:
# Evaluate the model's performance by comparing the predicted labels to the actual labels
accuracy = accuracy_score(y_test, y_pred)
print("Accuracy:", accuracy)

Accuracy: 0.7362637362637363


Now we use only GHG Emission (g) to make predictions

In [9]:
# Load the data into a pandas DataFrame
df_ghg = pd.read_csv('C:/Users/smvan/CFFS-S23/CFFS-22-23/data/final/2022_2023_CFFS_Outcomes/Data_Labelled_Gather22-23_with_name' +
                 '.csv', usecols=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14])
df_ghg.head()

Category            Displayed Name   ProdId                  Description  \
0  Caliente            Black bean soup  R-54947      CAL|Black bean soup|AAD   
1  Caliente             Guacamole bowl  R-55154       CAL|Bowl|Guacamole|AAD   
2  Caliente    Havana Add Chicken Bowl  R-62018  CAL|Bowl|Havana|Add Chicken   
3  Caliente    Havana Add Scallop Bowl  R-64909  CAL|Bowl|Havana|Add Scallop   
4  Caliente    Havana Plant-based Bowl  R-35896  CAL|Bowl|Havana|Plant based   

      SalesGroup  Weight (g)  GHG Emission (g)  N lost (g)  \
0  GV - CALIENTE       405.0        642.872067    3.699893   
1  GV - CALIENTE       345.0        577.312317    3.412346   
2  GV - CALIENTE       430.0       1120.107243   15.532360   
3  GV - CALIENTE       390.0        742.333777    6.220997   
4  GV - CALIENTE       415.0        769.178688    2.802186   

   Freshwater Withdrawals (L)  Stress-Weighted Water Use (L)  \
0                       53.60                        1723.18   
1                      115.97                        3039.52   
2                      125.86                         764.45   
3                       83.43                         677.50   
4                       83.83                         673.26   

   GHG Emission (g) / 100g  N lost (g) / 100g  \
0                   158.73               0.91   
1                   167.34               0.99   
2                   260.49               3.61   
3                   190.34               1.60   
4                   185.34               0.68   

   Freshwater Withdrawals (L) / 100g  Stress-Weighted Water Use (L) / 100g  \
0                              13.23                                425.48   
1                              33.61                                881.02   
2                              29.27                                177.78   
3                              21.39                                173.72   
4                              20.20                                162.23   

  GHG Only Label  
0          Green  
1          Green  
2         Yellow  
3         Yellow  
4         Yellow

In [10]:
# Split the dataset into input features (X) and target variable (y)
X = df_ghg[['GHG Emission (g)']]
y = df_ghg['GHG Only Label']

In [11]:
# Split the dataset into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

In [12]:
# Select the machine learning algorithm and fit the model to the training data
clf = DecisionTreeClassifier()
clf.fit(X_train, y_train)

DecisionTreeClassifier()

In [13]:
# Use the trained model to make predictions on the test dataset
y_pred = clf.predict(X_test)

In [14]:
# Evaluate the model's performance by comparing the predicted labels to the actual labels
accuracy = accuracy_score(y_test, y_pred)
print("Accuracy:", accuracy)

Accuracy: 0.6373626373626373


From data above we can see that we get a higher accuracy score in predictions using GHG Emission (g), N lost (g), Freshwater Withdrawals (L), Stress-Weighted Water Use (L) as compared to just using GHG Emission (g). (0.7362637362637363 > 0.6373626373626373)